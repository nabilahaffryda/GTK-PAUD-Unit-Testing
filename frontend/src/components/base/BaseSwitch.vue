<template>
  <v-chip class="ma-2" large :color="`${isTutup ? '' : !isAktif ? 'error' : 'primary'} lighten-5`">
    <v-tooltip top>
      <template v-slot:activator="{ on }">
        <v-btn
          class="mr-2 elevation-0"
          :color="`${isTutup ? '' : !isAktif ? 'error' : 'primary'} ${model === '' ? 'darken-2' : 'lighten-3'}`"
          :disabled="isTutup"
          fab
          x-small
          dark
          v-on="on"
          @click="select('')"
        >
          <v-icon>mdi-cancel</v-icon>
        </v-btn>
      </template>
      <span>Belum Diset</span>
    </v-tooltip>
    <v-tooltip top>
      <template v-slot:activator="{ on }">
        <v-btn
          class="mr-2 elevation-0"
          :color="`${isTutup ? '' : !isAktif ? 'error' : 'primary'} ${model === '1' ? 'darken-2' : 'lighten-3'}`"
          :disabled="isTutup"
          fab
          x-small
          dark
          v-on="on"
          @click="select('1')"
        >
          <v-icon>mdi-check</v-icon>
        </v-btn>
      </template>
      <span>Set True</span>
    </v-tooltip>
    <v-tooltip top>
      <template v-slot:activator="{ on }">
        <v-btn
          class="mr-2 elevation-0"
          :color="`${isTutup ? '' : !isAktif ? 'error' : 'primary'} ${model === '0' ? 'darken-2' : 'lighten-3'}`"
          :disabled="isTutup"
          fab
          x-small
          dark
          v-on="on"
          @click="select('0')"
        >
          <v-icon>mdi-close</v-icon>
        </v-btn>
      </template>
      <span>Set False</span>
    </v-tooltip>
  </v-chip>
</template>

<script>
export default {
  name: 'BaseSwitch',
  props: {
    value: {
      type: String,
      default: '',
    },
    isAktif: {
      type: Boolean,
      default: false,
    },
    isTutup: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      model: this.value || '',
    };
  },
  methods: {
    mutate() {
      this.$emit('input', this.model);
    },
    select(value) {
      this.model = value;
    },
  },
  watch: {
    value: function (value) {
      this.model = value;
    },
    model: 'mutate',
  },
};
</script>

<style scoped></style>
