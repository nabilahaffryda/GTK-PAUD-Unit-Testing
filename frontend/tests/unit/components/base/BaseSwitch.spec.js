import { mount, createLocalVue } from "@vue/test-utils";
import BaseSwitch from "@/components/base/BaseSwitch.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';
import VueRouter from 'vue-router';

Vue.use(Vuetify)
const localVue = createLocalVue();
localVue.use(VueRouter)
const router = new VueRouter();

describe('BaseSwitch.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })

    function wrapperFactory({ } = {}) {
        return mount(BaseSwitch, {
            localVue,
            vuetify,
            router,
        });
    }
    test('call mutate method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.mutate = jest.fn();
        wrapper.vm.mutate();
        expect(wrapper.vm.mutate.mock.calls.length).toBe(1);
    })
    test('call select method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.select = jest.fn();
        wrapper.vm.select();
        expect(wrapper.vm.select.mock.calls.length).toBe(1);
    })
})