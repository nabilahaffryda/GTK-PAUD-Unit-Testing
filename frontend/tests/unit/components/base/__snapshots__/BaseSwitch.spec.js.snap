// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSwitch.vue render switch 1`] = `<span class="ma-2 v-chip theme--light v-size--large error lighten-5"><span class="v-chip__content"><span class="v-tooltip v-tooltip--top"><!----><button type="button" class="mr-2 elevation-0 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round theme--dark v-size--x-small error darken-2"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-cancel theme--dark"></i></span></button></span> <span class="v-tooltip v-tooltip--top"><!----><button type="button" class="mr-2 elevation-0 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round theme--dark v-size--x-small error lighten-3"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-check theme--dark"></i></span></button></span> <span class="v-tooltip v-tooltip--top"><!----><button type="button" class="mr-2 elevation-0 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round theme--dark v-size--x-small error lighten-3"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close theme--dark"></i></span></button></span></span></span>`;
